ruma-signatures/src/functions.rs

```diff
@@ -731,13 +744,75 @@ pub fn redact(
     Ok(event)
 }
 
+/// Extracts the server names to check signatures for given event. It will return the sender's
+/// server (unless it's a third party invite) and the event id server (on v1 and v2 room versions)
+fn servers_to_check_signatures(
+    object: &CanonicalJsonObject,
+    version: &RoomVersionId,
+) -> Result<HashSet<String>, Error> {
```

Comment:
Why `String` instead of `ServerNameBox`? Also, have you considered using `BTreeSet`? For small amounts of items I think it should perform better (and it also has deterministic iteration order, which may or may not be relevant).